Fill array

AlexHalbesleben opened the pull request:
Resolves #5230

I'm not sure if this is something that should be implemented or not. Also, I'm wasn't entirely sure what to name the function, so I went with `fillArray`. If someone can think of a better name, let's change it.

imadx commented:
Since we have `_.repeat` for strings, do you think `arrayRepeat` would make sense as a function name?

imadx commented on fillArray.js:
```diff
@@ -0,0 +1,18 @@
+ * @param {Array} array The array to repeat
+ * @param {Number} times An integer specifying how many times elements of the existing array should be repeated
+ * @returns {Array} an array containing the elements of the existing array repeated
+ * @example fillArray([1, 2, 3], 3) // returns [1, 2, 3, 1, 2, 3, 1, 2, 3]
```
We seem to have the examples after a new line, in our other examples. Should we do the same to be consistent with the rest?

imadx commented on fillArray.js:
```diff
@@ -0,0 +1,18 @@
+ * @returns {Array} an array containing the elements of the existing array repeated
+ * @example fillArray([1, 2, 3], 3) // returns [1, 2, 3, 1, 2, 3, 1, 2, 3]
+ */
+function fillArray(array, times) {
```
We might need to validate the type for the inputs. Should we add tests for invalid inputs?

imadx replied:
Also `if` we are renaming this to `arrayRepeat`, we might need to update the local variable names also

imadx commented on fillArray.js:
```diff
@@ -0,0 +1,18 @@
+  return filledArray
+}
+
+export default fillArray
```
Missing a newline at the end?

AlexHalbesleben commented:
I've renamed the function and corresponding files to `arrayRepeat`. I also added type-checking (and the corresponding tests). 

I noticed that `repeat` might actually be able to work for arrays as well as strings with some minor changes. Do you think it's a better idea to merge `arrayRepeat` and `repeat`?

imadx commented:
> I noticed that `repeat` might actually be able to work for arrays as well as strings with some minor changes. Do you think it's a better idea to merge `arrayRepeat` and `repeat`?

Do you think it'll be a performance hit on the existing usages?

imadx commented on arrayRepeat.js:
```diff
@@ -0,0 +1,29 @@
+/**
+ * Creates a new array containing the elements of an existing array repeated
+ * @since 4.0.0
```
This would also need to be updated I think, not sure what the correct version should be though.

DoctypeRosenthal commented:
What about `repeatArray()`?
